src/alert-manager/src/cluster-utilization/send_alert.py

```diff
@@ -32,143 +31,183 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def datetime_to_hours(dt):
+    """Converts datetime.timedelta to hours
+
+    Parameters:
+    -----------
+    dt: datetime.timedelta
+
+    Returns:
+    --------
+    float
+    """
+    return dt.days * 24 + dt.seconds / 3600
+
+
 @enable_request_debug_log
-def get_job_usage_info(job_usage_result, rest_url):
+def get_usage_info(job_usage_result, user_usage_result, rest_url):
     job_infos = {}
-    job_usage = []
-    job_list = []
+    user_infos = {}
     # get all jobs
     headers = {'Authorization': "Bearer {}".format(TOKEN)}
     resp = requests.get(rest_url, headers=headers)
     resp.raise_for_status()
     job_list = resp.json()
 
+    for v in user_usage_result["data"]["result"]:
+        user_infos[v["metric"]["username"]] = {
+            "username": v["metric"]["username"],
+            "usage": v["value"][1][:6] + "%", "resources_occupied": 0
+        }
     for v in job_usage_result["data"]["result"]:
-        job_infos[v["metric"]["job_name"]] = {"usage": v["value"][1][:6] + "%"}
-    for job_name, job_info in job_infos.items():
-        url = urllib.parse.urljoin(rest_url + "/", job_name)
-        resp = requests.get(url, headers=headers)
-        if not resp.ok:
-            logging.warning("request failed %s", resp.text)
+        job_name = v["metric"]["job_name"]
+
+        matched_job = list(
+            filter(lambda job: "{}~{}".format(job["username"], job["name"]) == job_name,
+            job_list))
+
+        # ingore unfounded jobs
+        if not matched_job:
+            logging.warning("Job %s not found.", job_name)
             continue
-        resp_json = resp.json()
-        # get job duration
-        if not resp_json["jobStatus"]["appLaunchedTime"]:
+        job_info = matched_job[0]
+
+        # ignore jobs not started
+        if not job_info["launchedTime"]:
```

Comment:
Please make sure this field is available for running jobs

Reply:
> Please make sure this field is available for running jobs

will be fixed in https://github.com/microsoft/pai/pull/5307